### PR TITLE
dtc: Remove support for common.dts

### LINKED ISSUE #15359: The docs incorrectly state that common.dts integrates with mcuboot
The documentation is stating something that is incorrect. It says that;

> The [file common.dts] conditionally includes a fragment for MCUboot chain-loading, located at dts/common/mcuboot.overlay, if CONFIG_BOOTLOADER_MCUBOOT is set.

But in reality common.dts no longer does anything and should be removed. Along with any documentation referencing it.

### PR DESCRIPTION
**Pull request by SebastianBoe**
Remove the common.dts file which has been unused for a year.

common.dts at one point allowed us to do this:

> #ifdef CONFIG_BOOTLOADER_MCUBOOT	
> #include "mcuboot.overlay"	
> #endif

but since DT lost access to Kconfig options it has been unused.

The overridable variable DTS_COMMON_OVERLAYS, which by default points
to common.dts, is also unused in-tree, and any out-of-tree usage can
be ported over to use DTC_OVERLAY_FILE instead, so we remove the
variable as well.

This simplifies the configuration system.

Also, remove a nearby sentence in the docs that was saying DT uses Kconfig symbols.

fixes #15359

### CODE REVIEW

**Review by pabigot — APPROVED** [review]

**Comment by erwango** [comment]
Adding @mbolivar as he has been using this file (when it was not empty) IIRC

File: cmake/dts.cmake [hunk 1]
```diff
@@ -17,7 +17,6 @@ set(GENERATED_DTS_BOARD_CONF      ${PROJECT_BINARY_DIR}/include/generated/genera
 set(DTS_POST_CPP                  ${PROJECT_BINARY_DIR}/${BOARD}.dts.pre.tmp)
 
 set_ifndef(DTS_SOURCE ${BOARD_DIR}/${BOARD}.dts)
-set_ifndef(DTS_COMMON_OVERLAYS ${ZEPHYR_BASE}/dts/common/common.dts)
```
Comment (mbolivar) [inline cmake/dts.cmake]: Please replace this with something that will let the user know what happened.

```
if(DEFINED DTS_COMMON_OVERLAYS)
  message(FATAL_ERROR "DTS_COMMON_OVERLAYS is no longer supported. Use DTS_OVERLAY_FILE instead."
endif()
```

**Review by mbolivar — CHANGES_REQUESTED** [review]

**Review by dbkinder — APPROVED** [review]
doc change LGTM

File: cmake/dts.cmake (hunk 1 above)
Comment (SebastianBoe) [reply cmake/dts.cmake]: Fixed

**Comment by SebastianBoe** [comment]
@mbolivar : Your feedback has been applied, please re-asses the review.

**Review by galak — APPROVED** [review]

**Review by mbolivar — APPROVED** [review]

**Pull request merged**